Relax loadExternalTokens requirements to allow loading without id_token (#7165)
AzureAD/microsoft-authentication-library-for-js: lib/msal-browser/docs/testing.md

## The loadExternalTokens() API

The `loadExternalTokens()` API allows the loading of id, access and refresh tokens to the MSAL cache, which can then be fetched using `acquireTokenSilent()`.

> **Note: This is an advanced feature that is intended for testing purposes in the browser environment only. We do not recommend using this in a production app. For E2E testing recommendations, please refer to our [TestingSample](../../../samples/msal-browser-samples/TestingSample) instead.**

Reviewer: Are will calling it `testing only` API?

> **Note: This is an advanced feature that is intended for testing purposes in the browser environment only. We do not recommend using this in a production app. For E2E testing recommendations, please refer to our [TestingSample](../../../samples/msal-browser-samples/TestingSample) instead.**

Author: Resolving offline.

The `loadExternalTokens()` API can be accessed by calling `getTokenCache()` on MSAL Browser's `PublicClientApplication` instance.

```js
const msalTokenCache = myMSALObj.getTokenCache();
msalTokenCache.loadExternalTokens(
    silentRequest,
    serverResponse,
    loadTokenOptions
);
```

`loadExternalTokens()` takes in a request of type `SilentRequest`, a response of type `ExternalTokenResponse`, and options of type `LoadTokenOptions`.

See the type definitions for each, which can be imported from `@azure/msal-browser`:

-   [`SilentRequest`](https://azuread.github.io/microsoft-authentication-library-for-js/ref/types/_azure_msal_browser.SilentRequest.html)
-   [`ExternalTokenResponse`](https://azuread.github.io/microsoft-authentication-library-for-js/ref/types/_azure_msal_browser.ExternalTokenResponse.html)
-   [`LoadTokenOptions`](https://azuread.github.io/microsoft-authentication-library-for-js/ref/types/_azure_msal_browser.LoadTokenOptions.html)

## Loading tokens

> You can provide any combination of id, access and refresh tokens for caching but at a minimum the `loadExternalTokens` API requires one of the following sets of input parameters to identify who the tokens belong to and be capable of storing them in the cache:

Reviewer: ```suggestion
You can provide any combination of id, access and refresh tokens for caching but at a minimum the `loadExternalTokens` API requires one of the following sets of input parameters to identify token associations and cache appropriately:
```

If we derive `account` here, we can even say `to identity the accounts and cache...`

> -   A `SilentRequest` object with account information, OR

Reviewer: Can we link here to the exact format of `AccountInfo` that we expect? Just link to the typedocs.

Author: Done

-   A `SilentRequest` object with the authority AND a `LoadTokenOptions` object with `clientInfo`, OR

> -   A `SilentRequest` object with the authority AND a server response object with `client_info`

Reviewer: Do we need to add reference to use cases where `client_info` is not available? I understand it is dependent on the Confidential client flow that precedes the side loading, it may help for comprehensive use case reference in this case.

Author: The line below this one mentions you can use idToken instead and 2 lines above mentions you can use AccountInfo instead.

Reviewer: I meant something else entirely. More about "If you are ADFS, you will not have ` client_info` etc" .. but on second thought, folks who use this feature are already not doing simple use cases. Lets leave this as is.
